src/PrestaShopBundle/DependencyInjection/Compiler/ModuleControllerRegisterPass.php

```diff
@@ -0,0 +1,107 @@
+use RecursiveIteratorIterator;
+use ReflectionClass;
+use RegexIterator;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * This class is responsible for retrieving controllers in the modules that inherit from FrameworkBundleAdminController
+ * and declare them as services, also adding the autoconfigure and autowire tags.
+ *
+ * This modification is necessary to initialize the globalContainer and thus be able to retrieve all services
+ * via the $this->get function in the modules.
+ */
+class ModuleControllerRegisterPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container): void
+    {
+        $activeModules = $container->getParameter('prestashop.active_modules');
+        $moduleDir = $container->getParameter('prestashop.module_dir');
+
+        foreach ($activeModules as $moduleName) {
+            $fileIterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($moduleDir . $moduleName));
+            $phpFiles = new RegexIterator($fileIterator, '/\.php$/');
+
+            foreach ($phpFiles as $file) {
+                $className = $this->getClassNameFromFile($file->getRealPath());
+
+                if ($className !== null && class_exists($className)) {
+                    $reflector = new ReflectionClass($className);
+
+                    if ($reflector->isSubclassOf(FrameworkBundleAdminController::class)) {
+                        $definition = $this->getServiceDefinition($container, $className);
+                        $definition->addTag('controller.service_arguments');
+                        $definition->setAutoconfigured(true);
+                        $definition->setAutowired(true);
+                    }
+                }
+            }
+        }
+    }
+
+    private function getServiceDefinition(ContainerBuilder $container, $className): Definition
+    {
+        if ($container->has($className)) {
+            return $container->getDefinition($className);
+        }
+
+        /** @var Definition $definition */
+        foreach ($container->getDefinitions() as $definition) {
+            if ($definition->getClass() === $className) {
+                return $definition;
+            }
+        }
+
+        return $container->register($className, $className);
+    }
+
+    private function getClassNameFromFile(string $filePath): ?string
+    {
+        $contents = file_get_contents($filePath);
+        $namespace = $className = '';
+
+        if (preg_match('/namespace\s+(.+?);/s', $contents, $matches)) {
+            $namespace = $matches[1];
+        }
+
+        if (preg_match('/class\s+(\w+)\s+extends\s+FrameworkBundleAdminController/', $contents, $matches)) {
+            $className = $matches[1];
+        }
+
+        return $namespace ? $namespace . '\\' . $className : null;
```

Comment:
```suggestion
        return !empty($namespace) && !empty($className) ? $namespace . '\\' . $className : null;
```
You need both values no? And maybe the `class_exists` can also be in this method